File: src/main/java/top/criwits/scaleda/verilog/psi/factory/nodes/module/ParameterDeclarationPsiNode.java
package top.criwits.scaleda.verilog.psi.factory.nodes.module;

import com.intellij.lang.ASTNode;
import org.antlr.intellij.adaptor.psi.ANTLRPsiNode;
import org.jetbrains.annotations.NotNull;
import top.criwits.scaleda.verilog.psi.TypedDeclaration;

public class ParameterDeclarationPsiNode extends ANTLRPsiNode
        implements TypedDeclaration {

    public ParameterDeclarationPsiNode(@NotNull ASTNode node) {
        super(node);
    }

    public String getTypeText() {
        return getFirstChild().getFirstChild().getText();
    }

}

File: src/main/scala/top/scaleda/systemverilog/psi/nodes/parameter/ParameterDeclarationPsiNode.scala
package top.scaleda
package systemverilog.psi.nodes.parameter

import systemverilog.psi.nodes.signal.SignalDeclarationPsiNode

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement
import com.intellij.psi.util.PsiTreeUtil
import top.scaleda.systemverilog.psi.nodes.StructureViewNode

import java.util

class ParameterDeclarationPsiNode(node: ASTNode) extends SignalDeclarationPsiNode(node) with StructureViewNode {
  override def getTypeText: String = {
    // val typeChildren: Array[PsiElement] =
    //   util.Arrays.copyOf(this.getChildren, this.getChildren.length - 1) // I don't have range; I don't have semicolon!
    // typeChildren.map(_.getText).foldLeft("")(_ + " " + _).trim
    node.getText.split(';').headOption.getOrElse("unknown")
  }

  def getListOfParamAssignments: Option[ListOfParamAssignmentsPsiNode] = {
    Option(PsiTreeUtil.getChildOfType(this, classOf[ListOfParamAssignmentsPsiNode]))
  }

  override def getElementName: String = getTypeText
}
